prover/utils/parallel/parallel.go

```diff
@@ -48,6 +48,8 @@ func Execute(nbIterations int, work func(int, int), maxCpus ...int) {
 		}
 
 		go func() {
+			runtime.LockOSThread()
```

Comment:
Here and elsewhere - why lock work to particular thread? Does it allow to use some caches more efficiently? In that case, I think a small comment would be useful.